Lecture6/Task1/HttpClientsImplementation/Utils/RequestExecutor.java

```diff
@@ -0,0 +1,45 @@
+package HttpClientsImplementation.Utils;
+
+import Constants.HttpClients;
+import Constants.HttpMethods;
+import HttpClientsImplementation.ApacheHttpClient;
+import HttpClientsImplementation.RestAssured;
+
+public class RequestExecutor {
+
+    private static final String ERROR_MESSAGE = "Something went wrong => RequestExecutor)";
+
+    public RequestExecutor() {
+    }
+
+    @SuppressWarnings("unchecked")
+    public <T> T executeAndGetResponse(HttpClients httpClients, HttpMethods httpMethods, RequestDataPreparatory requestData) {
+        switch (httpClients) {
```

Comment:
such code is hard to maintain as adding new methods and clients cause changes in the main code of executing. It smells with rigidity